unit_tests/test_redis_client_base.py

```diff
@@ -0,0 +1,217 @@
+
+    def test_check_buffer_message_matching_command_and_id(self, redis_client_auto_register):
+        """checks the buffer message matching the command and id and returns as None"""
+        message = {"data": {"transaction_id": TRANSACTION_ID}}
+        redis_client_auto_register._blocking_command_responses \
+            = {"register": {"transaction_id": f"{TRANSACTION_ID}"}}
+        assert redis_client_auto_register \
+                   ._check_buffer_message_matching_command_and_id(message["data"]) is None
+
+    def test_check_buffer_message_matching_command_and_id_throws_exception(self, redis_client_auto_register):
+        """checks buffer message matching command and id does
+           contain transaction id throws exception"""
+        message = {}
+        with pytest.raises(RedisAPIException,
+                           match="The answer message does not contain a valid "
+                                 "\'transaction_id\' member."):
+            redis_client_auto_register._check_buffer_message_matching_command_and_id(message)
+
+    def test_check_buffer_message_matching_command_and_id_throws_another_exception(self, redis_client_auto_register):
+        """checks buffer message matching command and id throws exception"""
+        message = {"data": {"transaction_id": TRANSACTION_ID}}
+        with pytest.raises(RedisAPIException,
+                           match="There is no matching command response"
+                                 " in _blocking_command_responses."):
+            redis_client_auto_register \
+                ._check_buffer_message_matching_command_and_id(message["data"])
+
+    def test_register(self, redis_client_auto_register):
+        """checks whether the redis.db.publish is called with correct arguments"""
+        data = {"name": AREA_ID, "transaction_id": AREA_ID}
+        redis_client_auto_register.redis_db.publish = MagicMock()
+        redis_client_auto_register.register(is_blocking=False)
+        redis_client_auto_register.redis_db.publish \
+            .assert_called_once_with(f"{AREA_ID}/register_participant", json.dumps(data))
+
+    def test_register_blocking_true_throws_exception(self, redis_client_auto_register):
+        """checks whether the if is blocking is set to true throws exception"""
+        with pytest.raises(RedisAPIException,
+                           match="API registration process timed out. "
+                                 "Server will continue processing "
+                                 "your request on the background "
+                                 "and will notify you as soon as the "
+                                 "registration has been completed."):
+            redis_client_auto_register.register(is_blocking=True)
+
+    def test_register_blocking_false_throw_exception(self, redis_client_auto_register):
+        """checks whether the if is blocking is set to false throws exception"""
```

Comment:
Here also, the cause is not that you set `is_blocking=True`, but that the user is already registered. Please adapt the function name and docstring. thanks!

Reply:
Also, since we're talking about this:
- please use imperative mood in the docstrings (as described in [PEP 257](https://www.python.org/dev/peps/pep-0257))
  > The docstring is a phrase ending in a period. It prescribes the function or method's effect as a command ("Do this", "Return that"), not as a description; e.g. don't write "Returns the pathname ...".
- start docstrings with a capital letter
- end docstrings with a dot.